### ISSUE TITLE
New projects with existing SCC status aren't picked up as having SCC

### ISSUE THREAD

**Issue by neurobashing** [Komodo/KomodoEdit#2846]
### Short Summary
If I have an existing source tree that has SCC active (eg, it's tracked by Git) then doing project->new project, I get a project without SCC enabled and I can't seem to turn it on inside Komodo without re-cloning the entire repo.

### Steps to Reproduce
1. Have a dir that is already tracked by an SCC (I only use Git but I assume it happens for all)
2. Create a new project for that dir
3. Look at the project preferences; there is no SCC info. You can't update it.
4. Attempt to checkout an existing branch; you can only clone.

### Expected results
Broadly, if there is a .git dir in the root of the project, the details should be populated.

### Actual results
I have to re-clone.

### Platform Information

Komodo IDE, version 11.0.0-alpha1, build 90396, platform macosx.
Built on Wed Aug 9 23:39:14 2017.

**Comment by Naatan** [comment]
This has been bugged for a long time, Komodo checks the project file itself to get the SCC status. The project file needs to be under SCC for it to return a status, thus for new projects it does not return a SCC status.

**Comment by cgchoffman** [comment]
> The project file needs to be under SCC for it to return a status

I don't think this is true.  My KomodoIDE project file is not under SCC but I get an SCC status.

**Comment by cgchoffman** [comment]
What is the point of that screen?  It seems like extra work to see what the Commit pane already tells you.

**Comment by cgchoffman** [comment]
![capture](https://user-images.githubusercontent.com/655154/29323773-c02163f0-8196-11e7-95f8-46262078ea53.PNG)
This does not seem like helpful information.  @neurobashing, do you use this a lot?  Or did you just happen to find that it was broken?

**Comment by neurobashing** [comment]
It has bothered me for a while; I'm just going through all the parts of 11 and seeing what's changed, what still bothers me, etc.

**Comment by cgchoffman** [comment]
@neurobashing and @Naatan, what I'm getting at is that it's not a helpful screen over other more readily available tools in Komodo.  Do we even need this?  I'm suggesting we remove it, though it would be good to figure out WHY it's happening.

**Comment by Naatan** [comment]
> I don't think this is true. My KomodoIDE project file is not under SCC but I get an SCC status.

It is true, I was just in the code. There might be something creative in the way the project file status is detected, but it's definitely checking against the project file.

I agree it probably makes sense to just get rid of this screen, it doesn't really add much value. We should maybe replace it with something that DOES add value (shows remotes, branches, etc.)

**Comment by cgchoffman** [comment]
> There might be something creative in the way the project file status is detected, but it's definitely checking against the project file.

My project file is not under SCC so it can't be checking mine.  Must be a fallback.

**Comment by neurobashing** [comment]
I would be happy with getting rid of it; simplification FTW. My main use case/want is that for a given project already under source control, Ctrl-K,K should not present me with a "checkout" dialog and ask me to fill in a URL to checkout the source.

**Comment by cgchoffman** [comment]
> Ctrl-K,K should not present me with a "checkout" dialog and ask me to fill in a URL to checkout the source.

Is that happening??  You never mentioned this.  That's definitely a big deal.

**Comment by neurobashing** [comment]
i though it was related to my original report; I guess it's different?. 

I have a project that is under Git, and when I do Ctrl-K,K I get this:
![screen shot 2017-08-15 at 12 33 08 pm](https://user-images.githubusercontent.com/59380/29325500-73b26460-81d7-11e7-9222-a6ecceb708fc.png)
When I want is just a menu of `git branch` output (perhaps with an option to include `-r` or `-a`).

**Comment by Naatan** [comment]
Our checkout dialog is exactly that. It's badly named for git's use case but it's not a bug. Well maybe a UX bug, but not a bug in our code. This goes back to SCC polish that we will get to in a later version.

We'll leave this ticket open to deal with the pref screen. Feel free to file a separate issue about the checkout window if you feel so inclined.

Thanks for the feedback! :)

**Comment by neurobashing** [comment]
will do, thanks for listening.

**Comment by cgchoffman** [comment]
I should have looked at command names.  Are you looking for `Source Code Control: Pull Repository` @neurobashing?  There is no key binding for that currently so you would need to add it in your prefs.

**Comment by Naatan** [comment]
He's looking to checkout a commit or branch. In git world "checkout" means checkout a specific reference on an existing repository. With most other VCS "checkout" means to "clone" a new repository. The terminology is needlessly confusing, but then git isnt known for choosing sensible command names.

**Comment by cgchoffman** [comment]
Understood.
